Lib/fontTools/feaLib/variableScalar.py

```diff
@@ -81,11 +88,17 @@ def value_at_location(self, location):
         values = list(self.values.values())
         return self.model.interpolateFromMasters(loc, values)
 
-    @property
+    @cached_property
     def model(self):
         locations = [dict(self._normalized_location(k)) for k in self.values.keys()]
-        return VariationModel(locations)
-
+        key = tuple([tuple(x.items()) for x in locations])
+        if key in model_pool:
+            return model_pool[key]
+        m = VariationModel(locations)
+        model_pool[key] = m
+        return m
+
+    @lru_cache
```

Comment:
Is `self.values` mutable somehow? If not, then this could also be just @cached_property